packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+      this.updateBreadcrumbState();
+    }
+  }
+
+  private updateBreadcrumbState = async () => {
+    let updatedState = this.state;
+
+    // Update the base address if it has changed.
+    const baseEntityName = this.currentPathnames[1];
+    if (!(updatedState.base.entityName === baseEntityName)) {
+      updatedState = {
+        ...updatedState,
+        base: {
+          entityName: baseEntityName,
+          displayName:
+            `${baseEntityName}`.charAt(0).toUpperCase() +
+            `${baseEntityName}s`.slice(1),
+          url: `/${this.currentPathnames.slice(0, 2).join('/')}`,
+          isLast: false,
+        },
+      };
+    }
+
+    // Reset the last fields for base and the whole breadcrumb state.
+    // Set the base isLast to false unless it is proven otherwise later.
+    updatedState.base.isLast = false;
+    if (updatedState.last.displayName !== 'N/A') {
+      updatedState = {
+        ...updatedState,
+        last: {
+          displayName: 'N/A',
+        },
+      };
+    }
+
+    // Loop through each entry in the path name before the last.
+    // We always skip 2 go ahead in steps of 2 as the we expect
+    // the format to be /{entity}/{entityId}.
+    let hierarchyCount = 0;
+    const pathLength = this.currentPathnames.length;
+    for (let index = 1; index < pathLength; index += 2) {
+      // Get the entity and the data stored on the entity.
+      let entity = this.currentPathnames[index];
+
+      const link = `/${this.currentPathnames.slice(0, index + 3).join('/')}`;
+
+      // Check we are not the end of the pathnames array.
+      if (index < pathLength - 1) {
+        // If the index does not already exist,
+        // create the entry for the index.
+        if (!updatedState.currentHierarchy[hierarchyCount]) {
+          // Insert the new breadcrumb information into the array.
+          updatedState.currentHierarchy.splice(hierarchyCount, 1, {
+            id: '',
+            pathName: entity,
+            displayName: 'N/A',
+            url: link,
+          });
+        }
+
+        // Get the information held in the state for the current path name.
+        const entityInfo = updatedState.currentHierarchy[hierarchyCount];
+
+        // Get the entity Id in the path to compare with the saved one in our state.
+        const entityId = this.currentPathnames[index + 1];
+
+        // Check if an entity id is present or if the id has changed since the last update to the state.
+        if (entityInfo.id.length === 0 || entityInfo.id !== entityId) {
+          // In general the API endpoint will be our entity name and
+          // the entity field we want is the NAME of the entity.
+          let apiEntity = entity;
+          let requestEntityField = 'NAME';
+
+          // There are some exceptions to this when handling the DIAMOND/DLS
+          // depending on if 'proposal' has been found in the current path.
+          if (this.currentPathnames.includes('proposal')) {
```

Comment:
This should be extracted out instead to the config file

Reply:
@louise-davies added the breadcrumb settings to datagateway-table-settings.json file and loaded settings into Redux state in https://github.com/ral-facilities/datagateway/pull/70/commits/13bb780a15cb01f28bc0f0377608a14988b652da.

Reply:
@louise-davies basic configuration settings for breadcrumb have been defined. I will address these in the [ADR section](https://github.com/ral-facilities/scigateway/wiki/Architectural-Decision-Records-(ADRs)).